### DISCUSSION TITLE
ControlNet error on --nowebui

### DISCUSSION THREAD

**Discussion by bigboss97** [lllyasviel/stable-diffusion-webui-forge#347] [Q&A]
I've recently set up Forge with adding extra path for controlNet models of my A1111. Everything works fine on webui.
I've a Python script which interacts with A1111 nowebui. Now I run Forge in --nowebui mode and run the Python script. I'm getting following errors:
```
Model loaded in 38.1s (load weights from disk: 1.2s, forge load real models: 34.8s, calculate empty prompt: 2.0s).
*** Error running process: E:\bin\forge_webui\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "E:\bin\forge_webui\webui\modules\scripts.py", line 803, in process
        script.process(p, *script_args)
      File "E:\bin\forge_webui\system\python\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "E:\bin\forge_webui\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 546, in process        self.bound_check_params(unit)
      File "E:\bin\forge_webui\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 519, in bound_check_params
        unit.processor_res = int(preprocessor.slider_resolution.gradio_update_kwargs.get('value', 512))
    AttributeError: 'NoneType' object has no attribute 'slider_resolution'

---
*** Error running process_before_every_sampling: E:\bin\forge_webui\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "E:\bin\forge_webui\webui\modules\scripts.py", line 835, in process_before_every_sampling
        script.process_before_every_sampling(p, *script_args, **kwargs)
      File "E:\bin\forge_webui\system\python\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "E:\bin\forge_webui\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 555, in process_before_every_sampling
        self.process_unit_before_every_sampling(p, unit, self.current_params[i], *args, **kwargs)
    KeyError: 0
```
I thought it didn't pick up my A1111 controlNet path and I copied the model to forge_webui\webui\models\ControlNet. But it didn't help. Forge --nowebui does generate image, but ignores the controlNet.
Any suggestion?

**Comment by DubProphet** [comment]
having this same issue and error trying to utilize xl canny and depth into a deforum animation.  what's strange is the very first time i tried, i got a successful render with both controlnets working, i made no changes to anything but from that point forward it has been broken.

**Reply by bigboss97** [reply]
>  what's strange is the very first time i tried, i got a successful render with both controlnets working, i made no changes to anything but from that point forward it has been broken.

In my blur memory, I thought I've seen mine working, too. I tried to run Forge from scratch, still no luck.

I found this post [#375](https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/375) which looks promising. I tried this with no success:
```
def read_image(img_path):
    img = cv2.imread(str(img_path))
    _, bytes = cv2.imencode(".png", img)
    encoded_image = base64.b64encode(bytes).decode("utf-8")
    return encoded_image

base64image = read_image(poseFile)

"image": base64image,
```

My initial code which works on A1111 (noui):
```
image= encode_image( poseFile)

	  "input_image": image,
	  "module": "none",
          "model": "control_openpose-fp16 [9ca67cc5]",
```

**Reply by bigboss97** [reply]
> My initial code which works on A1111 (noui):
> 
> ```
> image= encode_image( poseFile)
> 
> 	  "input_image": image,
> 	  "module": "none",
>           "model": "control_openpose-fp16 [9ca67cc5]",
> ```
I spent two days and tried everything, now it works 😊
In Forge, it's "image", instead of "input_image" and "None" must be uppercase.
